gh_folder_download/main.py

```diff
@@ -82,30 +95,63 @@ def download_folder(
     force: bool,
 ) -> None:
     """
-    Download all contents at server_path with commit tag sha in
-    the repository.
+    Downloads a folder from a repository.
+
+    Args:
+        repository (Repository): The repository to download from.
+        sha (str): The SHA of the commit or tree to download.
+        path (str): The path of the folder to download.
+        output (Path): The output path where the folder will be saved.
+        force (bool): If True, existing folder will be removed before downloading.
+
+    Returns:
+        None: This function does not return anything.
     """
     fullpath = join(output, path)
+    existing_items = set(os.listdir(fullpath)) if os.path.exists(fullpath) else set()
+
+    # Get contents of the folder and assign to extra_items
+    contents = repository.get_dir_contents(path, ref=sha)
+    github_items = {content.name for content in contents}
+    extra_items = existing_items - github_items
+
     if exists(fullpath):
-        if force:
+        if extra_items:
```

Comment:
I think that if the `force` is `true` the removal of the folder should be automatic, if it's `false` then ask to user if he/she wants to delete the folder and continue or cancel the operation. So:

```py
if force:
    ...
elif extra_items:
    ...
else:
    ...
```